MKL_README.md

```diff
@@ -17,3 +17,106 @@ Installing and enabling the full MKL installation enables MKL support for all op
+6. If you have Intel Math Kernel Library (MKL) installed, set ```MKL_ROOT``` to point to ```MKL``` directory that contains the ```include``` and ```lib```. Typically, you can find the directory in
+```C:\Program Files (x86)\IntelSWTools\compilers_and_libraries_2018\windows\mkl```.
+7. If you don't have the Intel Math Kernel Library (MKL) installed, download and install [OpenBlas](http://sourceforge.net/projects/openblas/files/v0.2.14/). Note that you should also download ```mingw64.dll.zip`` along with openBLAS and add them to PATH.
+8. Set the environment variable ```OpenBLAS_HOME``` to point to the ```OpenBLAS``` directory that contains the ```include``` and ```lib``` directories. Typically, you can find the directory in ```C:\Program files (x86)\OpenBLAS\```. 
+
+After you have installed all of the required dependencies, build the MXNet source code:
+
+1. Download the MXNet source code from [GitHub](https://github.com/dmlc/mxnet). Don't forget to pull the submodules:
+```
+    git clone https://github.com/apache/incubator-mxnet.git ~/mxnet --recursive
+```
+
+2. Update mkldnn to the newest:
+```
+    cd 3rdparty/mkldnn/ && git checkout master && git pull
+```
+
+Or you can follow the [#216](https://github.com/intel/mkl-dnn/pull/216) to do some changes directly.
+
+3. Download [MKLML small library](https://github.com/intel/mkl-dnn/releases/download/v0.13/mklml_win_2018.0.2.20180127.zip):
+
+Extract it to `3rdparty/mkldnn/external` manually.
+
+4. Copy file `3rdparty/mkldnn/config_template.vcxproj` to incubator-mxnet root.
+
+5. modify mxnet CMakeLists:
+
+disable cuda and cudnn if you don't have cuda library and enable MKLDNN
+
+```
+mxnet_option(USE_CUDA             "Build with CUDA support"   OFF)
+mxnet_option(USE_CUDNN            "Build with cudnn support"  OFF) 
+mxnet_option(USE_MKLDNN           "Use MKLDNN variant of MKL (if MKL found)" ON IF USE_MKL_IF_AVAILABLE)
+mxnet_option(ENABLE_CUDA_RTC      "Build with CUDA runtime compilation support" OFF)
+```
+
+add line `add_definitions(-DMXNET_USE_MKLDNN=1)` so that it can build with openblas.
+
+```
+if(USE_MKL_IF_AVAILABLE)
+  if(USE_MKLDNN)
+    add_subdirectory(3rdparty/mkldnn)
+    include_directories(3rdparty/mkldnn/include)
+    list(APPEND mxnet_LINKER_LIBS mkldnn)
+    add_definitions(-DMXNET_USE_MKLDNN=1)
+  endif()
+  find_package(MKL)
+```
+
+6. Modify `incubator-mxnet\src\operator\tensor\elemwise_sum.h`:
+
+Modify `Sum` in `line 40,73,80,88,94,97` to `Sum2` since it has conflicts with `Sum` in MKLDNN.
+
+7. Start a Visual Studio command prompt.
+8. Use [CMake](https://cmake.org/) to create a Visual Studio solution in ```./build``` or some other directory. Make sure to specify the architecture in the 
+[CMake](https://cmake.org/) command:
+```
+    mkdir build
+    cd build
+    cmake -G "Visual Studio 14 Win64" ..
+```
+
+Note that you should close the openmp since MSVC doesn't support openMP3.0. Enable MKLDNN with `MKLDNN_VERBOSE=1`.
```

Comment:
do we need to turn off openmp in cmake?
why do we enable MKLDNN_VERBOSE here?

Reply:
our cmakefile should detect this automatically and disable the features accordingly. We should not expect users to change the configuration just in order to get it compiled locally. In the end, that's the purpose of cmake, right?